### ISSUE TITLE
Migration of the possibility to define CKEditor buttons per property for Sulu 2.0

### ISSUE THREAD

**Issue by simonnorris-publishit** [sulu/sulu#4898]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes
| Sulu Version | 2.0.1

#### Migration of the possibility to define CKEditor buttons per property
In Sulu 1.6, we had the possibility to define the buttons that are available per property (in the content type xml file). For Sulu 2.0, we have the possibility to modify which buttons show for the text_editor but only globally.
Is it possible to migrate this function to the new working of CKEditor 5?

**Comment by simonnorris-publishit** [comment]
I think the format you have suggested is exactly what we're looking for.
If you mean by 'options', the buttons that can be added/excluded, here's my list : 
- Bold
- Italic
- Underline
- Strike
- Superscript
- Subscript
- JustifyLeft
- JustifyCenter
- JustifyRight
- JustifyBlock
- NumberedList
- BulletedList
- Link
- Source

Where I fear it might be more complicated is when there are additional plugins added.
For example, with a Sulu 1.6 project, we added the 'mediaembed' plugin and declared it only for a few properties. Can this be easily added too?

**Comment by danrot** [comment]
Thanks for the list!

Regarding the third party thing: That's indeed not easily possible at the moment. We have a component that can load different adapters, but the [value is currently hardcoded](https://github.com/sulu/sulu/blob/46ca38a/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/TextEditor.js#L12).

But when thinking about it this would still not be enough, because the `TextEditorContainer` only takes certain props... I guess I'll have to think about that a bit more.

**Comment by alexander-schranz** [comment]
I would not go with configurate everything over the xml in the template. Thats mostly to much copy & paste for the developer in the xmls. I would more go with adding additional parameter to our CKEditor registries to allow different configrations so in the xml you only need to define something like:

```xml
<property name="title" type="texteditor">
     <params>
           <param name="config" value="minimal" />
    </params>
</property>
```

If we want we could make configrations configureable over the sulu_admin configs but I would not put more into the page xmls.

**Comment by danrot** [comment]
I don't think that this is a good idea... It is harder to understand what `minimal` is resp. where it comes from. And if you need multiple different text editors that will result in loads of different config values, quickly getting out of hand and with strange namings.

What I could imagine though is to be able to set default properties valid for all text editors in this installation, so that the amount of configuration can be minimized.

**Comment by danrot** [comment]
I have already mentioned it in #5214, but I am also going to mention it here for completeness: WDYT about handling more complicated configuration via x-includes? This would solve the problem not only for text editors, but for any content/field type with a more complex configuration.

**Comment by TheCadien** [comment]
I would like to push these Issue.

**Comment by danrot** [comment]
Do you mean by that, that you would like to help implement it? :slightly_smiling_face:

**Comment by niklasnatter** [comment]
I like the solution outlined in #5214 because it does not require us to create a mapping between the XML parameters and the CKEditor configuration and it is compatible with additionally installed CKEditor plugins. 

I am aware that it comes with the cost of needing to build the administration interface by yourself. But I could imagine to include commonly used configurations (like the one described in #5214) into the core and make them usable without creating the build by yourself.

**EDIT:** We could even allow to define configurations in PHP and add them to the `ckeditorConfigRegistry` when the administration interface is loaded at a later point. That would you would not need to build the administration interface by yourself.

**Comment by alexander-schranz** [comment]
As it is very similar I will close this issue in favor of: https://github.com/sulu/sulu/issues/5214
